apis/python/src/tiledb/vector_search/object_api/embeddings_ingestion.py

```diff
@@ -0,0 +1,307 @@
+
+    def submit_local(d, func, *args, **kwargs):
+        # Drop kwarg
+        kwargs.pop("image_name", None)
+        kwargs.pop("resources", None)
+        return d.submit_local(func, *args, **kwargs)
+
+    def create_dag(
+        object_index: ObjectIndex,
+        embeddings_uri: str,
+        external_ids_uri: str,
+        partitions: List[ObjectPartition],
+        object_partitions_per_worker: int,
+        object_work_tasks: int,
+        metadata_array_uri: str = None,
+        index_timestamp: int = None,
+        workers: int = -1,
+        worker_resources: Dict = None,
+        worker_image: str = DEFAULT_IMG_NAME,
+        verbose: bool = False,
+        trace_id: Optional[str] = None,
+        mode: Mode = Mode.LOCAL,
+        config: Optional[Mapping[str, Any]] = None,
+        namespace: Optional[str] = None,
+    ) -> dag.DAG:
+        if mode == Mode.BATCH:
+            d = dag.DAG(
+                name="embedding-generation",
+                mode=Mode.BATCH,
+                max_workers=workers,
+                retry_strategy=models.RetryStrategy(
+                    limit=1,
+                    retry_policy="Always",
+                ),
+            )
+        else:
+            d = dag.DAG(
+                name="embedding-generation",
+                mode=Mode.REALTIME,
+                max_workers=workers,
+                namespace="default",
+            )
+
+        submit = partial(submit_local, d)
+        if mode == Mode.BATCH or mode == Mode.REALTIME:
+            submit = d.submit
+
+        task_id = 0
+        num_partitions = len(partitions)
+        for i in range(0, num_partitions, object_partitions_per_worker):
+            start = i
+            end = i + object_partitions_per_worker
+            if end > num_partitions:
+                end = num_partitions
+            partition_dicts = []
+            for partition in partitions[start:end]:
+                partition_dicts.append(partition.init_kwargs())
+            submit(
+                compute_embeddings_udf,
+                object_reader_source_code=object_index.object_reader_source_code,
+                object_reader_class_name=object_index.object_reader_class_name,
+                object_reader_kwargs=object_index.object_reader_kwargs,
+                object_embedding_source_code=object_index.embedding_source_code,
+                object_embedding_class_name=object_index.embedding_class_name,
+                object_embedding_kwargs=object_index.embedding_kwargs,
+                partition_dicts=partition_dicts,
+                embeddings_uri=embeddings_uri,
+                external_ids_uri=external_ids_uri,
+                metadata_array_uri=metadata_array_uri,
+                index_timestamp=index_timestamp,
+                verbose=verbose,
+                trace_id=trace_id,
+                config=config,
+                name="generate_embeddings-" + str(task_id),
```

Comment:
(Minor) Other places use only hyphens `name="k-means-part-" + str(task_id),`:
```suggestion
                name="generate-embeddings-" + str(task_id),
```

Reply:
Done